application/app/services/upload/upload_files_provider.rb

```diff
@@ -18,10 +18,8 @@ def recent_files
 
         start_time = to_time(data.file.start_date)
         created_time = to_time(data.file.creation_date)
-        [
-          group,
-          group == 0 ? start_time : (group == 1 ? created_time : start_time)
-        ]
+        time_value = group == 0 ? start_time : (group == 1 ? created_time : start_time)
+        [group, -time_value.to_i]
```

Comment:
why us `-` in. `-time_value.to_i` ? shouldn't this be a positive number?

Reply:
This is to make the latest file come first and match the Downloads page order.
This way is a better UX as the in progress always come first, then pending, then the rest in descending order of completion.